src/symtab2gb/symtab2gb_parse_options.cpp

```diff
@@ -81,11 +84,18 @@ static void run_symtab2gb(
       throw invalid_source_file_exceptiont{
         "failed to typecheck symbol table from file '" + symtab_filename + "'"};
     }
-    goto_modelt goto_model{};
-    goto_model.symbol_table = symtab;
-    goto_convert(goto_model, message_handler);
-    link_goto_model(linked_goto_model, goto_model, message_handler);
+
```

Comment:
Am I right in thinking this is actually a larger change?  As I understand it instead of converting each `symbol_tablet` to a `goto_functionst` and linking the `goto_functionst` it is linking the `symbol_tablet`s and then doing one conversion.  Is this correct?  I think this might be a better way of doing things.

Reply:
Yes, you are absolutely right. I should have said that this PR should be reviewed commit-by-commit (and I hope the commit message of the first commit 3325c3f actually explains this).

Reply:
Thanks for the explanation.